app/controllers/articles_controller.rb

```diff
@@ -7,8 +7,9 @@ class ArticlesController < ApplicationController
   def index
     @articles = Article.all.includes(:user)
 
-    @articles = @articles.from_author(params[:author]) if params[:author].present?
+    @articles = @articles.authored_by(params[:author]) if params[:author].present?
     @articles = @articles.tagged_with(params[:tag]) if params[:tag].present?
+    @articles = @articles.favorited_by(params[:favorited]) if params[:favorited].present?
     @articles = @articles.limit(params[:limit] || 20).offset(params[:offset] || 0).order(created_at: :desc)
```

Comment:
essa lógica aqui dá pra melhorar pra não ficar tão repetitiva, saca só o exemplo desse cara aqui: https://www.justinweiss.com/articles/search-and-filter-rails-models-without-bloating-your-controller/